ci: Allow PIP_PACKAGES on centos

maflcko opened the pull request:
This was added in 7fc5e865b93af59364e9c8bf75ec68b4decc7e5d but I can't see a reason why this should be forbidden.

This is also needed for other changes (bumping the minimum python version).

Commit:
ci: Allow PIP_PACKAGES on centos

This was added in 7fc5e865b93af59364e9c8bf75ec68b4decc7e5d but I can't
see a reason why this should be forbidden.

hebasto commented on ci/test/00_setup_env_i686_centos.sh:
```diff
@@ -9,7 +9,8 @@ export LC_ALL=C.UTF-8
 export DOCKER_NAME_TAG=quay.io/centos/centos:stream8
-export DOCKER_PACKAGES="gcc-c++ glibc-devel.x86_64 libstdc++-devel.x86_64 glibc-devel.i686 libstdc++-devel.i686 ccache libtool make git python3 python3-zmq which patch lbzip2 xz procps-ng dash rsync coreutils bison"
+export DOCKER_PACKAGES="gcc-c++ glibc-devel.x86_64 libstdc++-devel.x86_64 glibc-devel.i686 libstdc++-devel.i686 ccache libtool make git python3 python3-pip zeromq which patch lbzip2 xz procps-ng dash rsync coreutils bison"
+export PIP_PACKAGES="pyzmq"
```
How does this PR change the "32-bit + dash [gui] [CentOS 8]" CI task?

Asking because the `interface_zmq.py` functional test [passes](https://cirrus-ci.com/task/5464815220031488) on the current master branch.

MarcoFalke replied:
It shouldn't. All it does is replace the `python3-zmq` system package, which has wrapped the `pyzmq` from pip and `zeromq`. It is replaced by an explicit installation of both "sub"-packages.

hebasto replied:
Then, probably, the relevant docs should be adjusted as well?

hebasto replied:
> All it does is replace the `python3-zmq` system package...

Other tasks still use `python3-zmq`.

MarcoFalke replied:
> apt-get install python3-zmq

centos is not using `apt-get`, so any doc update seems unrelated. If your system provides `python3-zmq` you can use it, however, you can also use `pip3 install pyzmq`, which should work everywhere.

hebasto replied:
> centos is not using `apt-get`, so any doc update seems unrelated

That is why docs need to be updated like that o similar:
```
The ZMQ functional test requires a python ZMQ library. To install it:
- on Debian/Ubuntu, run `sudo apt-get install python3-zmq`
- on other systems, run `pip3 install pyzmq`
```

But it is unrelated to this PR CI changes themself.

---

> All it does is replace the `python3-zmq` system package, which has wrapped the `pyzmq` from pip and `zeromq`. It is replaced by an explicit installation of both "sub"-packages.

And what are benefits of that?

hebasto replied:
> Also, it is needed for other changes (bumping the minimum python version).

Perhaps, mention this in the PR description?

MarcoFalke replied:
> Perhaps, mention this in the PR description?

Thx, done

MarcoFalke commented on ci/test/04_install.sh:
```diff
@@ -78,9 +72,16 @@ elif [ "$CI_USE_APT_INSTALL" != "no" ]; then
+  else
     # shellcheck disable=SC2086
-    ${CI_RETRY_EXE} pip3 install --user $PIP_PACKAGES
+    ${CI_RETRY_EXE} CI_EXEC pip3 install --user $PIP_PACKAGES
```
Review note: Adding CI_EXEC is required, otherwise it would result in a failure.

To reproduce the failure:


```
$ RESTART_CI_DOCKER_BEFORE_RUN=1 FILE_ENV="./ci/test/00_setup_env_i686_centos.sh" ./ci/test_run_all.sh
...
bash: line 1: pip3: command not found

MarcoFalke replied:
This also fixes a bug on historic commits. To reproduce the bug on the historic commits:

```
$ git checkout 3c3bd9022026a75e15492ba9cf8bf3b72866b9a9~ && RESTART_CI_DOCKER_BEFORE_RUN=1 FILE_ENV="./ci/test/00_setup_env_i686_multiprocess.sh" ./ci/test_run_all.sh
...
bash: line 1: pip3: command not found

hebasto reviewed (approved):
ACK fa684bcde97e7ed5de102429e1e656c6d091a547, I have reviewed the code and it looks OK, I agree it can be merged.

MarcoFalke commented:
cc @fanquake (author of 7fc5e865b93af59364e9c8bf75ec68b4decc7e5d)

fanquake reviewed (approved):
ACK fa684bcde97e7ed5de102429e1e656c6d091a547 - I see lief still being installed.

Commit:
ci: Remove unused package

Address feedback from https://github.com/bitcoin/bitcoin/pull/24561/files#r985719812

MarcoFalke commented on ci/test/00_setup_env_i686_centos.sh:
```diff
@@ -9,7 +9,8 @@ export LC_ALL=C.UTF-8
 export CONTAINER_NAME=ci_i686_centos
 export DOCKER_NAME_TAG=quay.io/centos/centos:stream8
-export DOCKER_PACKAGES="gcc-c++ glibc-devel.x86_64 libstdc++-devel.x86_64 glibc-devel.i686 libstdc++-devel.i686 ccache libtool make git python3 python3-zmq which patch lbzip2 xz procps-ng dash rsync coreutils bison"
+export DOCKER_PACKAGES="gcc-c++ glibc-devel.x86_64 libstdc++-devel.x86_64 glibc-devel.i686 libstdc++-devel.i686 ccache libtool make git python3 python3-pip zeromq which patch lbzip2 xz procps-ng dash rsync coreutils bison"
```
(fixed in last force push)

MarcoFalke commented:
For reference, interface_zmq.py:

* on master macos: https://cirrus-ci.com/task/6731452615229440?logs=ci#L2918
* on master centos: https://cirrus-ci.com/task/5464815220031488?logs=ci#L3322
* here macos: https://cirrus-ci.com/task/5487962711392256?logs=ci#L2918
* here centos: https://cirrus-ci.com/task/6121281408991232?logs=ci#L3303

hebasto reviewed (approved):
re-ACK fa6054e952f4522b98dc89609033950a3cbfd06c
